From 6c3f19ba5b6db287980241e4d20be9ac5d0c1293 Mon Sep 17 00:00:00 2001
From: Li Dongyang <dongyang.li@anu.edu.au>
Date: Mon, 16 May 2016 17:27:23 +1000
Subject: [PATCH 3/8] LU-8056 socklnd: NETIF_F_ALL_CSUM renamed to
 NETIF_F_CSUM_MASK

In kernel 4.5 NETIF_F_CSUM_MASK got renamed to NETIF_F_CSUM_MASK.
This patch handles the name change.

Linux-commit:a188222b6ed29404ac2d4232d35d1fe0e77af370

Signed-off-by: Li Dongyang <dongyang.li@anu.edu.au>
Change-Id: Id57505eeca613303c584d3cf74284920a837bb43
---
 lnet/klnds/socklnd/socklnd.h     | 4 ++++
 lnet/klnds/socklnd/socklnd_lib.c | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/lnet/klnds/socklnd/socklnd.h b/lnet/klnds/socklnd/socklnd.h
index f02003a..54794c0 100644
--- a/lnet/klnds/socklnd/socklnd.h
+++ b/lnet/klnds/socklnd/socklnd.h
@@ -66,6 +66,10 @@
 	tcp_sendpage(sk, page, offset, size, flags)
 #endif /* HAVE_TCP_SENDPAGE_USE_SOCKET */
 
+#ifndef NETIF_F_CSUM_MASK
+# define NETIF_F_CSUM_MASK NETIF_F_ALL_CSUM
+#endif
+
 /* assume one thread for each connection type */
 #define SOCKNAL_NSCHEDS		3
 #define SOCKNAL_NSCHEDS_HIGH	(SOCKNAL_NSCHEDS << 1)
diff --git a/lnet/klnds/socklnd/socklnd_lib.c b/lnet/klnds/socklnd/socklnd_lib.c
index 9dca0e4..e26db71 100644
--- a/lnet/klnds/socklnd/socklnd_lib.c
+++ b/lnet/klnds/socklnd/socklnd_lib.c
@@ -71,7 +71,7 @@ ksocknal_lib_zc_capable(ksock_conn_t *conn)
 
 	/* ZC if the socket supports scatter/gather and doesn't need software
 	 * checksums */
-	return ((caps & NETIF_F_SG) != 0 && (caps & NETIF_F_ALL_CSUM) != 0);
+	return ((caps & NETIF_F_SG) != 0 && (caps & NETIF_F_CSUM_MASK) != 0);
 }
 
 int
-- 
2.8.2

